services/user-service/utils/passport.ts

```diff
@@ -0,0 +1,127 @@
+import passport from 'passport';
+import LocalStrategy from 'passport-local';
+import FacebookStrategy from "passport-facebook";
+import { UserAuthTypeEnum } from '../utils/enum';
+import User from '../models/user-model';
+import UserAuth from '../models/user-auth-model';
+import sequelize from './sequelize';
+import { Transaction } from 'sequelize';
+
+const strategy = {
+  login: async (profileId: string, authType: UserAuthTypeEnum, options: {
+    user: {
+      firstName: string,
+      lastName?: string | null | undefined,
+      email: string,
+      imageUrl?: string | null | undefined,
+      auth: {
+        profileId: string,
+        accessToken?: string | undefined, 
+        refreshToken?: string| undefined,
+        data?: object | undefined,
+      }
```

Comment:
move this `user` type to its own definition and use it in the `services/user-service/actions/auth-action/facebook/redirect-action.ts`